lib/personalized_search/product_preferences_manager.dart

```diff
@@ -103,16 +103,16 @@ class ProductPreferencesManager {
   }
 
   /// Returns whether an attribute is important as per the user preferences.
-  bool isAttributeImportant(String attributeId) {
+  bool? isAttributeImportant(String attributeId) {
     final String importanceId = getImportanceIdForAttributeId(attributeId);
     final int? importanceIndex =
         _availablePreferenceImportances?.getImportanceIndex(importanceId);
-    if (importanceIndex == null ||
-        importanceIndex == PreferenceImportance.INDEX_NOT_IMPORTANT) {
-      // Not important.
-      return false;
+    if (importanceIndex == null) {
```

Comment:
sugg: Check `_availablePreferenceImportances == null` instead, since that's the only reason why `importanceIndex` might be null.

Reply:
Ah but we don't know if that's true in the future, maybe `getImportanceIndex` will return `null` in another case later.

I think the point of the cl was to stay consistent with the other APIs that only return null if `getImportanceIndex` returns null

Reply:
`int getImportanceIndex(...)` can never return `null` :)

Reply:
Ack, feel free to send a follow up PR, I'm not sure it matters much :)